packages/builder/builder-rspack-provider/src/plugins/swc.ts

```diff
@@ -1,20 +1,49 @@
-import * as path from 'path';
-
 import {
-  BuilderTarget,
+  JS_REGEX,
+  TS_REGEX,
+  mergeRegex,
+  applyScriptCondition,
   getBrowserslistWithDefault,
-  logger,
+  BuilderTarget,
   setConfig,
   isWebTarget,
+  BundlerChain,
   addCoreJsEntry,
+  BundlerChainRule,
+  logger,
 } from '@modern-js/builder-shared';
+import { cloneDeep } from '@modern-js/utils/lodash';
+import * as path from 'path';
 import type {
   BuilderPlugin,
-  BuilderPluginAPI,
   NormalizedConfig,
-  RspackConfig,
+  NormalizedSourceConfig,
+  BuiltinSwcLoaderOptions,
 } from '../types';
-import { Builtins } from '@rspack/core';
+
+export function getDefaultSwcConfig(): BuiltinSwcLoaderOptions {
+  return {
+    jsc: {
+      externalHelpers: true,
+      parser: {
+        tsx: true,
+        syntax: 'typescript',
+        decorators: true,
+      },
+      // Avoid the webpack magic comment to be removed
+      // https://github.com/swc-project/swc/issues/6403
+      preserveAllComments: true,
+    },
+    isModule: 'unknown',
+    minify: false, // for loader, we don't need to minify, we do minification using plugin
+    sourceMaps: true,
+    env: {
+      targets: '> 0.01%, not dead, not op_mini all',
```

Comment:
Should we use the browserslist config here?

Reply:
done